### PR TITLE
AUTH-298

### PR DESCRIPTION
**Pull request by luizjaccao**
## Description
Adds multi-line capacity to Tooltip component.

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] IE 11
  - [ ] Firefox
  - [x] Chrome
  - [ ] Safari
  - [ ] Mobile
- [x] Ux review validation

### CODE REVIEW

**Comment by luizjaccao** [comment]
Before:
image
After:
image

File: components/Tooltip/Tooltip.jsx [hunk 1]
```diff
@@ -43,17 +43,18 @@ const Tip = styled.div`
 
 const TipText = styled.span`
   display: inline-block;
-  max-width: 250px;
```
Comment (luizjaccao) [inline components/Tooltip/Tooltip.jsx:46]: This line has been removed for the Tooltip component not to truncate text.

File: components/Tooltip/Tooltip.jsx [hunk 2]
```diff
@@ -43,17 +43,18 @@ const Tip = styled.div`
...
   overflow: hidden;
   text-overflow: ellipsis;
-  white-space: nowrap;
+  white-space: pre;
```
Comment (luizjaccao) [inline components/Tooltip/Tooltip.jsx:48]: This line has been added for the Tootip component accept `\n` as enter.

File: components/Tooltip/Tooltip.jsx [hunk 3]
```diff
@@ -43,17 +43,18 @@ const Tip = styled.div`
...
   line-height: 20px;
+  text-align: left;
```
Comment (luizjaccao) [inline components/Tooltip/Tooltip.jsx:50]: This line has been added to left align the text.

File: components/Tooltip/Tooltip.jsx [hunk 4]
```diff
@@ -43,17 +43,18 @@ const Tip = styled.div`
...
 `;
 
 const Wrapper = styled.div`
   position: relative;
   float: left;
   clear: left;
+  width: max-content;
```
Comment (luizjaccao) [inline components/Tooltip/Tooltip.jsx:57]: This line has been added to adjust the size of Tooltip to the size of text.

File: components/Tooltip/Tooltip.jsx (hunk 1 above)
Comment (tcelestino) [reply components/Tooltip/Tooltip.jsx:46]: what do you think to switch `max-width` to `min-width`? If you use it, you don't need remove this and you don't need to set `max-content` in line https://github.com/catho/quantum/pull/233/files#r431827523

File: components/Tooltip/Tooltip.jsx (hunk 2 above)
Comment (tcelestino) [reply components/Tooltip/Tooltip.jsx:48]: what do you think to use `pre-wrap` instead?

File: components/Tooltip/Tooltip.jsx (hunk 3 above)
Comment (tcelestino) [reply components/Tooltip/Tooltip.jsx:50]: what do you think to set this property using theme? That's can be used to _left_ or _center_ text.

File: components/Tooltip/Tooltip.jsx [hunk 5]
```diff
@@ -12,6 +12,7 @@ const Tip = styled.div`
   border-radius: 4px;
   font-weight: bold;
   opacity: ${({ visible }) => (visible ? '1' : '0')};
+  visibility: ${({ visible }) => (visible ? 'visible' : 'hidden')};
```
Comment (luizjaccao) [inline components/Tooltip/Tooltip.jsx:15]: This line has been added to Tooltip only appear when mouse is over the wrapper.

File: components/Tooltip/Tooltip.jsx (hunk 1 above)
Comment (luizjaccao) [reply components/Tooltip/Tooltip.jsx:46]: min-width change the expected layout 
image

File: components/Tooltip/Tooltip.jsx (hunk 2 above)
Comment (luizjaccao) [reply components/Tooltip/Tooltip.jsx:49]: without `min-width` this change breaks the layout

File: components/Tooltip/Tooltip.jsx (hunk 1 above)
Comment (pedrosouza) [reply components/Tooltip/Tooltip.jsx:46]: This change is breaking the layout:
image

Before the change:
image

Comment (pedrosouza) [reply components/Tooltip/Tooltip.jsx:46]: Why don't we add a prop to change the behavior for the case that you are addressing and keep the original behavior when not using this prop?

**Comment by rapahaeru** [comment]
Suggestions:

- creates a new Boolean prop called `multiline` (example). If this props coming `true`, in the componente will have a condicional its implements the new needs. Keeping the original behavior when the prop was `false`.
-   when the new behavior was added, does not forget to implements its regression and unit tests for the new prop.

What do you think?

File: components/Tooltip/Tooltip.jsx [hunk 6]
```diff
@@ -87,7 +91,7 @@ class Tooltip extends Component {
           visible={visibleProp || visibleState}
           theme={theme}
         >
-          <TipText>{text}</TipText>
+          <TipText multiline={multiline}>{text}</TipText>
```
Comment (rapahaeru) [inline components/Tooltip/Tooltip.jsx:94]: How about of creates a unit test for this new prop?

File: components/Tooltip/Tooltip.jsx [hunk 7]
```diff
@@ -117,6 +122,7 @@ Tooltip.propTypes = {
 Tooltip.defaultProps = {
   placement: 'top',
   visible: false,
+  multiline: false,
```
Comment (rapahaeru) [inline components/Tooltip/Tooltip.jsx:125]: How about of create a new example, using this new prop, in story of storybook, for documentation?

Comment (rapahaeru) [inline components/Tooltip/Tooltip.jsx:125]: I think we should implement a new regression test using this new prop.

**Pull request merged**